Add initial import for containers

mjrider opened the pull request:
TODO:
- [x] figure out how to import the mandatory image variable, ideas are welcome
- [x] fix Read function to read all parameters
- [x] add documentation

jtopjian commented:
This is a tough one.

The best case scenario is if we can find _something_ via the API that references the image. I did a cursory look (I can look more indepth if needed) and found a few references, but oddly nothing that references what we normally use in the `image` parameter.

The alternative is to pass in metadata via the command-line and parse it in the import function:

```shell
$ terraform import lxd_container.foo foo/ubuntu
```

Then do a split on `/` and use `ubuntu` as the image info.  This is a fairly common thing to do for importing... it's just kind of ugly.

mjrider commented:
@jtopjian so that would make the 'ID" syntax for example
remote:container/images:alpine/3.5/amd
i think that would work just fine
the options are then
- container
- remote:container
- container/image:description
- remote:container/image:description

and the parse rules:
- split in max 2 parts on /
- part1:
  - parse with lxd parse function like now
- part2:
  - set as image if is set


i already added the feature that in the special case that 'image' is set to "" (which happens now ) the diff functions ignores it, and terraform doesn't plan to destroy the container because of changes

jtopjian commented:
All sounds good to me.

Also, I just realized, we should update the README to document that importing exists for certain resources.

mjrider commented:
@jtopjian what do you think of the following documentation snipped

Importing existing containers is possible with the following ID syntax
terraform import lxd_container.container [remote:]name[/images:alpine/3.5].
where:
- remote, optional,  is the name of the remote in the provider config
- name, required,  is the container name
- /images:alpine/3.5, optional,  translates to `image = images:alpine/3.5` in the resource configuration

jtopjian commented:
@mjrider That sounds great.

mjrider commented:
@jtopjian @sl1pm4t did you have time to review this PR ?

jtopjian commented on README.md:
```diff
@@ -143,6 +143,13 @@ resource "lxd_container" "test1" {
+
+remote, optional, is the name of the remote in the provider config
+name, required, is the container name
+/images:alpine/3.5, optional, translates to image = images:alpine/3.5 in the resource configuration
```
I think this should be moved to the "Reference" area of the documentation rather than in the "Example" area. Perhaps a new section called "Importing" that is similar to [this](https://github.com/sl1pm4t/terraform-provider-lxd/blob/master/README.md#container-network-access).

Also, if you look at the rendered README, the last three lines are all placed on a single line. I think they should be formatted as a list in Markdown.

jtopjian commented on lxd/import_resource_lxd_container_test.go:
```diff
@@ -0,0 +1,56 @@
+
+			resource.TestStep{
+				ResourceName: resourceName,
+				//				Config:            testAccContainer_basic(containerName),
```
nit: this looks weird.

jtopjian commented on lxd/resource_lxd_container.go:
```diff
@@ -344,12 +348,28 @@ func resourceLxdContainerRead(d *schema.ResourceData, meta interface{}) error {
+		} else if strings.HasPrefix(k, "security.") {
+			config[k] = v
+		} else if strings.HasPrefix(k, "user.") {
+			config[k] = v
```
Is it possible to use a `switch` statement here? It's usually cleaner to use a switch instead of multiple `else if` statements. If switch isn't possible, no problem.

mjrider replied:
afaik i can't do a switch with the hasPrefix call

jtopjian replied:
Yes, you are absolutely correct.

jtopjian commented on lxd/resource_lxd_container.go:
```diff
@@ -581,6 +601,43 @@ func resourceLxdContainerExists(d *schema.ResourceData, meta interface{}) (exist
+		d.Set("name", name)
+	} else {
+		return nil, err
+	}
```
Could the above be rewritten as:

```go
ct, _, err := server.GetContainerState(name)
if err != nil {
    return nil, err
}

if ct == nil {
    return nil, fmt.Errorf("Unable to get container state")
}

log.Printf("[DEBUG] Import container state %#v", ct)
d.SetId(name)
d.Set("name", name)
```

jtopjian commented:
@mjrider Sorry for the delay.

Thank you for your work and patience with this one. It probably wasn't the easiest to implement, so your work is appreciated :)
